### PR TITLE
Feature fix duplicated definitions

### LINKED ISSUE #811: Repeated definitions of register macros
Various definitions related with kratos variables are repeated through the code:

Specifically, identical definitions for:

- `KRATOS_REGISTER_IN_PYTHON_VARIABLE`
- `KRATOS_REGISTER_IN_PYTHON_3D_VARIABLE_WITH_COMPONENTS`

exists in:

- `define.h`
- `add_c2c_variables_to_python.cpp`
- `add_cfd_variables_to_python.cpp`
- `add_containers_to_python.cpp`
- `add_dem_variables_to_python.cpp`
- `add_deprecated_variables_to_python.cpp`
- `add_fsi_variables_to_python.cpp`
- `add_legacy_variables_to_python.cpp`
- `add_mapping_variables_to_python.cpp`
- `add_mat_variables_to_python.cpp`

This may have been the result of result for a problem while registering variables that no longer seems to happen. Maybe just with C2C.

My proposal is to remove the definitions from all files except for `add_c2c_variables_to_python.cpp` and change that one so it uses its own definitions in case is necessary instead of redefining the global ones.

### PR DESCRIPTION
**Pull request by roigcarlo**
This fixes #811 and #812 plus a little cleanup of unnecessary includes in that part of the code.

### CODE REVIEW

File: kratos/includes/define.h [hunk 1]
```diff
@@ -349,13 +349,13 @@ catch(...) { Block KRATOS_THROW_ERROR(std::runtime_error, "Unknown error", MoreI
 #undef KRATOS_REGISTER_IN_PYTHON_FLAG_IMPLEMENTATION
 #endif
 #define KRATOS_REGISTER_IN_PYTHON_FLAG_IMPLEMENTATION(flag) \
- scope().attr(#flag) = boost::ref(flag)      \
+ scope().attr(#flag) = boost::ref(flag);      \
```
Comment (pooyan-dadvand) [inline kratos/includes/define.h]: Do we need this semicolon? In my opinion we should leave this without and put it when it uses to make format corrector work well.

File: kratos/includes/define.h [hunk 2]
```diff
@@ -349,13 +349,13 @@ catch(...) { Block KRATOS_THROW_ERROR(std::runtime_error, "Unknown error", MoreI
...
  
 #ifdef KRATOS_REGISTER_IN_PYTHON_FLAG
 #undef KRATOS_REGISTER_IN_PYTHON_FLAG
 #endif
 #define KRATOS_REGISTER_IN_PYTHON_FLAG(flag) \
-    KRATOS_REGISTER_IN_PYTHON_FLAG_IMPLEMENTATION(flag);   \
```
Comment (pooyan-dadvand) [inline kratos/includes/define.h]: And here is needed for the same reason.

**Review by pooyan-dadvand — CHANGES_REQUESTED** [review]
I agree with changes but not the semicolon modifications.

**Comment by roigcarlo** [comment]
@pooyan-dadvand I understand the problem with the semicolon but the two versions were in the code before and the one you suggest makes the code in `add_containets_to_python.cpp` fail.

Best I can do is to move the semicolon to both statements in the usage macro like this:

```C++
#define KRATOS_REGISTER_IN_PYTHON_FLAG(flag) \
KRATOS_REGISTER_IN_PYTHON_FLAG_IMPLEMENTATION(flag);   \
KRATOS_REGISTER_IN_PYTHON_FLAG_IMPLEMENTATION(NOT_##flag);
```

Is that ok?

**Comment by pooyan-dadvand** [comment]
I would remove the final semicolon to be added in the file the macro used.

**Review by pooyan-dadvand — APPROVED** [review]
👍

**Comment by pooyan-dadvand** [comment]
Please wait until checks are passed.

**Pull request merged**